pkg/util/slices/map.go

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package slices are utilities to deal with slices
+package slices
+
+// Map returns a new slice with the result of applying fn to each element.
+func Map[S ~[]E, E any, RE any](s S, fn func(E) RE) []RE {
```

Comment:
There is a `Map` implementation proposed in https://github.com/golang/go/issues/61898 that may be more idiomatic.

Reply:
Those seem to use iterators which require Go 1.23

Reply:
There is a separate PR out to upgrade to go 1.23: https://github.com/DataDog/datadog-agent/pull/31022

Reply:
We can always change the implementation and usage once that is merged. I don't want to tie this PR to the Go upgrade for such a minor part of it.

Reply:
this is merged now btw